src/main/kotlin/kweb/Kweb.kt

```diff
@@ -169,45 +169,39 @@ class Kweb private constructor(
     There is a bit of duplication here, because we have to pass in the javascript string separately.
     We need it to create the debugToken. Some server2ClientMessages will contain the javascript,
      but messages that call cached functions will not have it. So we have to make sure we pass it in separately.*/
-    fun callJs(server2ClientMessage: Server2ClientMessage, javascript: String) {
-        val wsClientData = clientState.getIfPresent(server2ClientMessage.yourId)
-                ?: error("Client id ${server2ClientMessage.yourId} not found")
+    fun callJs(sessionId: String, funcCall: FunctionCall, javascript: String) {
+        val wsClientData = clientState.getIfPresent(sessionId)
+                ?: error("Client id $sessionId not found")
         wsClientData.lastModified = Instant.now()
         val debugToken: String? = if(!debug) null else {
             val dt = abs(random.nextLong()).toString(16)
             wsClientData.debugTokens[dt] = DebugInfo(javascript, "executing", Throwable())
             dt
         }
-        server2ClientMessage.debugToken = debugToken
+        funcCall.debugToken = debugToken
         val outboundMessageCatcher = outboundMessageCatcher.get()
         if (outboundMessageCatcher == null) {
-            wsClientData.send(server2ClientMessage)
+            wsClientData.send(Server2ClientMessage(sessionId, funcCall))
         } else {
-            logger.debug("Temporarily storing message for ${server2ClientMessage.yourId} in threadlocal outboundMessageCatcher")
-            val jsFunction = JsFunction(server2ClientMessage.jsId!!, server2ClientMessage.arguments!!)
+            logger.debug("Temporarily storing message for $sessionId in threadlocal outboundMessageCatcher")
+            val jsFunction = JsFunction(funcCall.jsId!!, funcCall.arguments)
```

Comment:
Are `JsFunction` and `FuncCall` similar enough that we could use the same class for both?

Reply:
That's a good thought. `JsFunction` only has 2 fields, and both of those are in `FuncCall`, so we should be able to just use `FuncCall`. That's an easy change for me to make.